src/allmydata/util/base62.py

```diff
@@ -101,10 +118,10 @@ def a2b_l(cs, lengthinbits):
         numvalues *= 62
 
     numvalues = 2**lengthinbits
-    bytes = []
+    result_bytes = []
     while numvalues > 1:
-        bytes.append(value % 256)
+        result_bytes.append(value % 256)
         value //= 256
         numvalues //= 256
 
-    return ''.join([chr(b) for b in reversed(bytes)]) # make it big-endian
+    return bytes([b for b in reversed(result_bytes)]) # make it big-endian
```

Comment:
This whole module I'll have to look at with fresh eyes in the morning, I guess.  The test suite still doesn't seem great.  There are ... zero? tests that validate anything except round-tripping and output length.

Reply:
I added some hard-coded known values, generated off of master.